scripts/ci/sync_extensions.py

```diff
@@ -62,27 +62,30 @@ def _sync_wheel(ext, updated_indexes, failed_urls, overwrite, temp_dir):
     if not overwrite:
         cmd = ['az', 'storage', 'blob', 'exists', '--container-name', f'{STORAGE_CONTAINER}', '--account-name',
                f'{STORAGE_ACCOUNT}', '--name', f'{blob_name}', '--auth-mode', 'login']
-        result = subprocess.run(cmd, capture_output=True)
-        if result.stdout and json.loads(result.stdout)['exists']:
+        try:
+            result = subprocess.run(cmd, capture_output=True, check=True)
+        except subprocess.CalledProcessError as ex:
+            raise Exception(f"Failed to check if '{blob_name}' exists in the storage") from ex
```

Comment:
If this `try ... catch ...` pattern is used multiple times, perhaps we can extract a function such as `az_cmd`? `azdev` has a `pip_cmd` function for similar purpose: https://github.com/Azure/azure-cli-dev-tools/blob/711ca36e3c3a76f4567467f4370b04dca5d7354c/azdev/utilities/command.py#L96

Also, instead of showing `Failed to check if '{blob_name}' exists in the storage` in case of failure, we can print `check if '{blob_name}' exists in the storage` unconditionally before the command is run.